bitcoin/psbt.c

```diff
@@ -80,6 +110,28 @@ struct wally_psbt_input *psbt_add_input(struct wally_psbt *psbt,
 	return &psbt->inputs[insert_at];
 }
 
+struct wally_psbt_input *psbt_append_input(struct wally_psbt *psbt,
+					   const struct bitcoin_txid *txid,
+					   u32 outnum, u32 sequence)
+{
+	struct wally_tx_input *tx_in;
+	struct wally_psbt_input *input;
+	size_t insert_at;
+
+	insert_at = psbt->num_inputs;
+
+	if (wally_tx_input_init_alloc(txid->shad.sha.u.u8,
+				      sizeof(struct bitcoin_txid),
+				      outnum, sequence, NULL, 0, NULL,
+				      &tx_in) != WALLY_OK)
+		abort();
+
+	tx_in->features = chainparams->is_elements ? WALLY_TX_IS_ELEMENTS : 0;
+	input = psbt_add_input(psbt, tx_in, insert_at);
```

Comment:
features is an internal impl detail and we may end up using it for more than just elements. please use wally_tx_elements_input_init_alloc for elements instead.

Reply:
ack, updated